samples/sensor/stream_fifo/src/main.c

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright (c) 2024 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/sys/util_macro.h>
+#include <zephyr/kernel.h>
+#include <zephyr/rtio/rtio.h>
+#include <zephyr/drivers/sensor.h>
+
+#define STREAMDEV_ALIAS(i) DT_ALIAS(_CONCAT(stream, i))
+#define STREAMDEV_DEVICE(i, _) \
+	IF_ENABLED(DT_NODE_EXISTS(STREAMDEV_ALIAS(i)), (DEVICE_DT_GET(STREAMDEV_ALIAS(i)),))
+#define NUM_SENSORS 1
+
+/* support up to 10 sensors */
+static const struct device *const sensors[] = { LISTIFY(10, STREAMDEV_DEVICE, ()) };
+
+#define STREAM_IODEV_SYM(id) CONCAT(accel_iodev, id)
+#define STREAM_IODEV_PTR(id, _) &STREAM_IODEV_SYM(id)
+
+#define STREAM_TRIGGERS					   \
+	{ SENSOR_TRIG_FIFO_FULL, SENSOR_STREAM_DATA_NOP }, \
+	{ SENSOR_TRIG_FIFO_WATERMARK, SENSOR_STREAM_DATA_INCLUDE }
+
+#define STREAM_DEFINE_IODEV(id, _)    \
+	SENSOR_DT_STREAM_IODEV(	      \
+		STREAM_IODEV_SYM(id), \
+		STREAMDEV_ALIAS(id),  \
+		STREAM_TRIGGERS);
+
+LISTIFY(NUM_SENSORS, STREAM_DEFINE_IODEV, (;));
+
+struct rtio_iodev *iodevs[NUM_SENSORS] = { LISTIFY(NUM_SENSORS, STREAM_IODEV_PTR, (,)) };
+
+RTIO_DEFINE_WITH_MEMPOOL(stream_ctx, NUM_SENSORS, NUM_SENSORS,
+			 NUM_SENSORS * 20, 256, sizeof(void *));
+
+struct sensor_chan_spec accel_chan = { SENSOR_CHAN_ACCEL_XYZ, 0 };
+struct sensor_chan_spec gyro_chan = { SENSOR_CHAN_GYRO_XYZ, 0 };
+struct sensor_chan_spec temp_chan = { SENSOR_CHAN_DIE_TEMP, 0 };
+struct sensor_chan_spec rot_vector_chan = { SENSOR_CHAN_GAME_ROTATION_VECTOR, 0 };
+struct sensor_chan_spec gravity_chan = { SENSOR_CHAN_GRAVITY_VECTOR, 0 };
+struct sensor_chan_spec gbias_chan = { SENSOR_CHAN_GBIAS_XYZ, 0 };
+
+static uint8_t accel_buf[128] = { 0 };
+static uint8_t gyro_buf[128] = { 0 };
+static uint8_t temp_buf[64] = { 0 };
+static uint8_t rot_vect_buf[128] = { 0 };
+static uint8_t gravity_buf[128] = { 0 };
+static uint8_t gbias_buf[128] = { 0 };
+
+static int print_accels_stream(const struct device *dev, struct rtio_iodev *iodev)
+{
+	int rc = 0;
+	const struct sensor_decoder_api *decoder;
+	struct rtio_cqe *cqe;
+	uint8_t *buf;
+	uint32_t buf_len;
+	struct rtio_sqe *handles[NUM_SENSORS];
+	struct sensor_three_axis_data *accel_data = (struct sensor_three_axis_data *)accel_buf;
+	struct sensor_three_axis_data *gyro_data = (struct sensor_three_axis_data *)gyro_buf;
+	struct sensor_q31_data *temp_data = (struct sensor_q31_data *)temp_buf;
+	struct sensor_three_axis_data *gravity_data = (struct sensor_three_axis_data *)gravity_buf;
+	struct sensor_three_axis_data *gbias_data = (struct sensor_three_axis_data *)gbias_buf;
+	struct sensor_game_rotation_vector_data *rot_vect_data =
+				(struct sensor_game_rotation_vector_data *)rot_vect_buf;
+
+	/* Start the streams */
+	for (int i = 0; i < NUM_SENSORS; i++) {
+		printk("sensor_stream\n");
+		sensor_stream(iodevs[i], &stream_ctx, NULL, &handles[i]);
+	}
+
+	while (1) {
```

Comment:
Could you clarify how is this loop supposed to work? The caller already has an infinite loop so I'm slightly confused on the intended behavior.

Reply:
There may be device-tree macrobatics to achieve it, but as it is the sample looks ok. Perhaps a future enhancement?

Reply:
Yes, it makes sense to explore more and enhance it in the future